File: src/security/lib/tpm/src/session.rs
// Copyright 2022 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

use tpm2_tss_sys as tss_sys;

/// Strong typing for a SessionHandle which is just a subtype of `ESYS_TR`.
#[derive(Debug, Copy, Clone, PartialEq, Eq)]
pub struct SessionHandle(tss_sys::ESYS_TR);

impl From<SessionHandle> for tss_sys::ESYS_TR {
    fn from(item: SessionHandle) -> tss_sys::ESYS_TR {
        item.0
    }
}

impl From<tss_sys::ESYS_TR> for SessionHandle {
    fn from(item: tss_sys::ESYS_TR) -> SessionHandle {
        SessionHandle(item)
    }
}

File: src/testing/end_to_end/honeydew/honeydew/interfaces/affordances/session.py
# Copyright 2023 The Fuchsia Authors. All rights reserved.
# Use of this source code is governed by a BSD-style license that can be
# found in the LICENSE file.
"""Abstract base class for Session affordance."""

import abc


class Session(abc.ABC):
    """Abstract base class for Session affordance."""

    @abc.abstractmethod
    def start(self) -> None:
        """Start session.

        Raises:
            honeydew.errors.SessionError: session failed to start.
        """

    @abc.abstractmethod
    def add_component(self, url: str) -> None:
        """Instantiates a component by its URL and adds to the session.

        Args:
            url: url of the component

        Raises:
            honeydew.errors.SessionError: Session failed to launch component
                with given url. Session is not started.
        """

    @abc.abstractmethod
    def stop(self) -> None:
        """Stop the session

        Raises:
            honeydew.errors.SessionError: Session failed to stop the session.
        """
